code/modules/events/queen_mothers_blessing.dm

```diff
@@ -0,0 +1,36 @@
+#define QM_HEAL_AMOUNT 250
+
+/datum/round_event_control/queen_mothers_blessing
+	name = "Queen Mother's Blessing"
+	typepath = /datum/round_event/queen_mothers_blessing
+	weight = 15
+	earliest_start = 5 MINUTES
+
+	gamemode_blacklist = list("Combat Patrol","Civil War","Sensor Capture")
+
+/datum/round_event_control/queen_mothers_blessing/can_spawn_event(players_amt, gamemode)
+	if(length(GLOB.alive_xeno_list) <= 0)
+		return FALSE
+	return ..()
+
+/datum/round_event/queen_mothers_blessing/start()
+	for(var/mob/living/carbon/xenomorph/X in shuffle(GLOB.alive_xeno_list))
+		if(isminion(X))
+			continue
+		bless_xeno(X)
+		break
+
+/datum/round_event/queen_mothers_blessing/proc/bless_xeno(mob/living/carbon/xenomorph/X)
+	X.evolution_stored = X.evolution_stored + (X.evolution_stored * 1.75)
+	if(!isxenolarva(X) || (X.tier != XENO_UPGRADE_FOUR)) //larva do not have proper caste datums, trying to force one results in a runtime
```

Comment:
```suggestion
	if(!isxenolarva(X) && (X.tier != XENO_UPGRADE_FOUR)) //larva do not have proper caste datums, trying to force one results in a runtime
```
|| would result in any xeno that is not a larva being forced to upgrade three, including primordials, aswell as larvae counting as viable target.

Reply:
Hmm true, should make this two if statements because if it's `&&` it'll never be true because you can't be primo and a larva at the same time if I'm not understanding it wrong.

Reply:
&& would mean "do thing if [not a larva] and also [not upgrade four]"
Aka
is larva + not upgrade four => 0
is larva + is upgrade four (somehow) => 0
not larva + not upgrade four => 1
not larva + is upgrade four => 0